src/JsonParseErrorException.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/*
+    Copyright 2014 Rustici Software
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+namespace TinCan;
+
+use Exception;
+
+class JsonParseErrorException extends Exception
+{
+
+    private static $msg = 'The provided json value "%s" couldn\'t be parsed. json_decode() provided the following error message: "%s" with the error code %d.';
+
+    private $malformedValue;
+    private $jsonErrorNumber;
+    private $jsonErrorMessage;
+
+    public function __construct($malformedValue, $jsonErrorNumber, $jsonErrorMessage, Exception $previous = null)
+    {
```

Comment:
Cool @brianjmiller it would be useful if it's one of [these](https://github.com/squizlabs/PHP_CodeSniffer/tree/master/CodeSniffer/Standards) so that we can automate the process of finding inconsistencies.